### PR TITLE
Make the `aoc join` command a slash command.

### LINKED ISSUE #64: AoC: Change leaderboard invite DM to ephemeral message slash command/button
Reliant on #19 being merged first.

I'd like to make the leaderboard invite code a slash command or semi-permanent button with an ephemeral message. This will allow people to keep their DMs closed and overall make the process smoother.

### PR DESCRIPTION
**Pull request by D0rs4n**
Closes #64

### CODE REVIEW

File: bot/exts/advent_of_code/_cog.py [hunk 1]
```diff
@@ -183,22 +190,25 @@ async def about_aoc(self, ctx: commands.Context) -> None:
         """Respond with an explanation of all things Advent of Code."""
         await ctx.send(embed=self.cached_about_aoc)
 
-    @commands.guild_only()
-    @adventofcode_group.command(name="join", aliases=("j",), brief="Learn how to join the leaderboard (via DM)")
+    @aoc_slash_group.command(name="join")
```
Comment (janine9vn) [inline bot/exts/advent_of_code/_cog.py:193]: We should add a description parameter here to give a better description for end users. Something like: "Get the join code for our community Advent of Code leaderboard"

**Review by janine9vn — APPROVED** [review]
Small comment, but I won't block because of it since it's ultra minor

File: bot/exts/advent_of_code/_cog.py (hunk 1 above)
Comment (D0rs4n) [reply bot/exts/advent_of_code/_cog.py]: Donem

File: bot/bot.py [hunk 2]
```diff
@@ -17,6 +20,7 @@ class SirRobin(BotBase):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
+        self.cog_load_task: asyncio.Task = None
```
Comment (ChrisLovering) [inline bot/bot.py:23]:
```suggestion
        self.cog_load_task: asyncio.Task | None = None
```

File: bot/exts/advent_of_code/_cog.py [hunk 3]
```diff
@@ -42,6 +43,12 @@ class AdventOfCode(commands.Cog):
     # RedisCache[member_id: None]
     completionist_block_list = RedisCache()
 
+    aoc_slash_group = app_commands.Group(
+        name="aoc",
+        description="All of the Advent of Code commands.",
+        guild_ids=[Client.guild]
```
Comment (ChrisLovering) [inline bot/exts/advent_of_code/_cog.py:49]:
```suggestion
        guild_ids=[Client.guild],
```

File: bot/exts/advent_of_code/_cog.py [hunk 4]
```diff
@@ -183,22 +190,25 @@ async def about_aoc(self, ctx: commands.Context) -> None:
...
         allowed_months = (Month.NOVEMBER.value, Month.DECEMBER.value)
         if not (
             current_date.month in allowed_months and current_date.year == AocConfig.year
             or current_date.month == Month.JANUARY.value and current_date.year == AocConfig.year + 1
         ):
             # Only allow joining the leaderboard in the run up to AOC and the January following.
-            await ctx.send(f"The Python Discord leaderboard for {current_date.year} is not yet available!")
+            await interaction.response.send_message(
+                f"The Python Discord leaderboard for {current_date.year} is not yet available!",
+                ephemeral=True
```
Comment (ChrisLovering) [inline bot/exts/advent_of_code/_cog.py:207]:
```suggestion
                ephemeral=True,
```

File: bot/exts/advent_of_code/_cog.py [hunk 5]
```diff
@@ -207,7 +217,10 @@ async def join_leaderboard(self, ctx: commands.Context) -> None:
             try:
                 join_code = await _helpers.get_public_join_code(author)
             except _helpers.FetchingLeaderboardFailedError:
-                await ctx.send(":x: Failed to get join code! Notified maintainers.")
+                await interaction.response.send_message(
+                    ":x: Failed to get join code! Notified maintainers.",
+                    ephemeral=True
```
Comment (ChrisLovering) [inline bot/exts/advent_of_code/_cog.py:222]:
```suggestion
                    ephemeral=True,
```

**Review by ChrisLovering — APPROVED** [review]

**Comment by ChrisLovering** [comment]
Just minor comments, feel free to merge with or without them

**Pull request merged**